Code/GraphMol/SmilesParse/test.cpp

```diff
@@ -4066,6 +4066,51 @@ void testGithub1925() {
   BOOST_LOG(rdInfoLog) << "done" << std::endl;
 }
 
+
+void testdoRandomSmileGeneration() {
+  BOOST_LOG(rdInfoLog) << "-------------------------------------\n";
+  BOOST_LOG(rdInfoLog)
+      << "Testing the random Generator for SMILES\n"
+      << std::endl;
+  {
+    std::srand(1); // be sure we use it for testcase!
+    const char *benzenes[1] = {
+        "c1ccncc1C"};
+
+    const char *rulesmiles[7] = {
+       "c1ccncc1C","c1cc(C)cnc1","c1ccc(C)cn1","n1cccc(C)c1",
+        "c1ncccc1C","c1(C)cccnc1","Cc1cccnc1"};
+    
+    const char *randomsmiles[7] = {
+      "c1ccncc1C","c1cc(cnc1)C","c1ccc(cn1)C","n1cccc(c1)C","c1ncccc1C",
+      "c1(cccnc1)C","Cc1cccnc1"};
+    
```

Comment:
In order to have the random numbers be reproducible, I think you need to call `std::srand()` here before you start calling `std::rand()` inside of `MolToSmiles()`.